middlewares/accesslog/logger.go

```diff
@@ -0,0 +1,147 @@
+package accesslog
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"sync/atomic"
+	"time"
+)
+
+type key int
+
+const (
+	// DataTableKey is the key within the request context used to
+	// store the Log Data Table
+	DataTableKey key = iota
```

Comment:
Is there a particular reason why we use 0/iota as the key here? Why not a more meaningful, human-readable string?

To the least, this would make debugging easier.